src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -1,5 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
```

Comment:
 MUST FIX: Deployment blocks *all* other commands until it's finished (thought we had fixed this).  Perhaps it needs a test.

Reply:
It blocks the entire language server, so that means all commands (e.g. create bicep config) as well as linting as colorization.  Here's one to play with to see what I mean.
```bicep
param cosmosDBDatabaseName string = 'db1'
param cosmosDBDatabaseThroughput int = 4000
var location = 'westus'

resource cosmosDBAccount 'Microsoft.DocumentDB/databaseAccounts@2021-03-15' = {
  name: uniqueString(resourceGroup().id)
  location: location
  kind: 'GlobalDocumentDB'
  properties: {
    consistencyPolicy: {
      defaultConsistencyLevel: 'Eventual'
      maxStalenessPrefix: 1
      maxIntervalInSeconds: 5
    }
    locations: [
      {
        locationName: location
        failoverPriority: 0
      }
    ]
    databaseAccountOfferType: 'Standard'
    enableAutomaticFailover: true
    capabilities: [
      {
        name: 'EnableTable'
      }
    ]
  }
}

resource cosmosDBDatabase 'Microsoft.DocumentDB/databaseAccounts/sqlDatabases@2020-04-01' = {
  parent: cosmosDBAccount
  name: cosmosDBDatabaseName // "The property \"name\" is declared multiple times in this object. Remove or rename the duplicate properties.","
  properties: {
    // The property "properties" is declared multiple times in this object. Remove or rename the duplicate properties.
    resource: {
      id: cosmosDBDatabaseName
    }
    options: {
      throughput: cosmosDBDatabaseThroughput
    }
  }
}
```

Reply:
(Delete it after deployment, they're expensive even if empty)

Reply:
Fixed here:
https://github.com/Azure/bicep/commit/84be0ce156d02f5e72934384fd055d86a57f5115

Thanks!